getopt.c

```diff
@@ -0,0 +1,69 @@
+#define _POSIX_C_SOURCE 200809L
+#include <mrsh/getopt.h>
+#include <stdio.h>
+
+char *mrsh_optarg = NULL;
+int mrsh_optind = 1;
+int mrsh_opterr = 1;
+int mrsh_optopt = 0;
+
+int mrsh_getopt(int argc, char *const argv[], const char *optstring) {
+	mrsh_optarg = NULL;
+
+	if (mrsh_optind >= argc) {
+		return -1;
+	}
+
+	if (argv[mrsh_optind] != NULL) {
+		if (argv[mrsh_optind][0] != '-') {
+			return -1;
+		}
+
+		if (argv[mrsh_optind][1] == '\0') {
+			return -1;
+		}
+
+		if (argv[mrsh_optind][1] == '-') {
+			mrsh_optind++;
```

Comment:
Oh, I see this here. I wonder if it would be possible to restructure this to be easier to follow...

Reply:
I could use `strncmp` rather than inspecting the elements individually which might make it more obvious. Any thoughts @emersion ?

Reply:
I don't have an opinion about this. Either is fine by me.